samples/Tutorials/CSharpIntermediate/CSharpIntermediate/CSharpIntermediate.Game/05_Async/AsyncWebApi.cs

```diff
@@ -12,43 +11,46 @@ namespace CSharpIntermediate.Code
 {
     public class AsyncWebApi : AsyncScript
     {
+        private List<OpenCollectiveEvent> openCollectiveEvents;
+
         public override async Task Execute()
         {
+            openCollectiveEvents = new List<OpenCollectiveEvent>();
 
             while (Game.IsRunning)
             {
-                DebugText.Print($"Press G to load Api data and Log it", new Int2(500, 200));
-                if (Input.IsKeyPressed(Stride.Input.Keys.G)){
+                int drawX = 500, drawY = 600;
+                DebugText.Print($"Press A to get Api data from https://opencollective.com/stride3d", new Int2(drawX, drawY));
+
+                if (Input.IsKeyPressed(Stride.Input.Keys.G))
+                {
                     await RetrieveStrideRepos();
                 }
 
+                foreach (var openCollectiveEvent in openCollectiveEvents)
```

Comment:
You might have to await NextFrame at the end of the `if` above to make sure to execute DebugText.Print on the game thread.

Reply:
@manio143 Hi!, thanks for the feedback. Would you add this specifically in the Retrieve method or right after the await? If I try this locally it don't see any behavior change.

Reply:
I would say after the await in this method so that only one method does sync with the frames. I don't expect there to be changes in behavior in general, but since the continuation of the async call for a Task may run on another thread and Stride's APIs aren't thread safe, you can sometimes get into a race condition.